test/irqbalance.bats

```diff
@@ -0,0 +1,198 @@
+		mv -v "$BANNEDCPUS_CONF".bkp "$BANNEDCPUS_CONF"
+	elif [ -f /tmp/.test_owns_bannedcpus_conf ]; then
+		rm -f "$BANNEDCPUS_CONF"
+		rm -f /tmp/.test_owns_bannedcpus_conf
+	else
+		echo "error: Something went really wrong with $BANNEDCPUS_CONF !"
+		exit 1
+	fi
+}
+
+@test "irqbalance cpu ban list save" {
+	# given
+	if ! grep -Eq '^[1,3,7,f]{1,}$' /proc/irq/default_smp_affinity; then
+		skip "requires default IRQ smp affinity (not banned CPUs)"
+	fi
+	[ -f "$CONFIGLET" ] && rm -f "$CONFIGLET"
+
+	[ -f "$BANNEDCPUS_CONF" ] && rm -f "$BANNEDCPUS_CONF"
+
+	local expected_banned_cpus
+	expected_banned_cpus=$(sed -n 's/^IRQBALANCE_BANNED_CPUS=\"\?\([^\"]*\)\"\?/\1/p' "$IRQBALANCE_CONF")
+	if [ -z "$expected_banned_cpus" ]; then
+		echo "error: unable to read 'IRQBALANCE_BANNED_CPUS' from ${IRQBALANCE_CONF}"
+		return 1
+	fi
+
+	# when
+	OVERRIDE_OPTIONS="--irqbalance-config-file ${IRQBALANCE_CONF}" start_crio
+
+	# then
+	if [ ! -f "$BANNEDCPUS_CONF" ]; then
+		echo "error: ${BANNEDCPUS_CONF} file should have been created by CRI-o"
+		return 2
+	fi
+
+	local banned_cpus
+	banned_cpus=$(cat "$BANNEDCPUS_CONF")
+
+	[ "$expected_banned_cpus" == "$banned_cpus" ]
+}
+
+@test "irqbalance cpu ban list restore - default" {
+	# given
+	if ! grep -Eq '^[1,3,7,f]{1,}$' /proc/irq/default_smp_affinity; then
+		skip "requires default IRQ smp affinity (not banned CPUs)"
+	fi
+	[ -f "$CONFIGLET" ] && rm -f "$CONFIGLET"
+
+	echo "IRQBALANCE_BANNED_CPUS=\"0\"" > "$IRQBALANCE_CONF"
+
+	local banned_cpus_for_conf
+	banned_cpus_for_conf=$(cat /proc/irq/default_smp_affinity)
+	echo "$banned_cpus_for_conf" > "$BANNEDCPUS_CONF"
+
+	# when
+	OVERRIDE_OPTIONS="--irqbalance-config-file ${IRQBALANCE_CONF}" start_crio
+
+	# then
+	local banned_cpus
+	banned_cpus=$(sed -n 's/^IRQBALANCE_BANNED_CPUS=\"\?\([^\"]*\)\"\?/\1/p' "$IRQBALANCE_CONF")
+
+	[ "$banned_cpus_for_conf" == "$banned_cpus" ]
+}
+
+# disable restore file, check it does NOT clear the irqbalance config
+@test "irqbalance cpu ban list restore - disable" {
+	# given
+	if ! grep -Eq '^[1,3,7,f]{1,}$' /proc/irq/default_smp_affinity; then
+		skip "requires default IRQ smp affinity (not banned CPUs)"
+	fi
+	[ -f "$CONFIGLET" ] && rm -f "$CONFIGLET"
+
+	local expected_banned_cpus
+	expected_banned_cpus=$(sed -n 's/^IRQBALANCE_BANNED_CPUS=\"\?\([^\"]*\)\"\?/\1/p' "$IRQBALANCE_CONF")
+	if [ -z "$expected_banned_cpus" ]; then
+		echo "error: unable to read 'IRQBALANCE_BANNED_CPUS' from ${IRQBALANCE_CONF}"
+		return 1
+	fi
+
+	[ -f "$BANNEDCPUS_CONF" ] && rm -f "$BANNEDCPUS_CONF"
+
+	# when
+	OVERRIDE_OPTIONS="--irqbalance-config-file ${IRQBALANCE_CONF} --irqbalance-config-restore-file disable" start_crio
+
+	# then
+	local banned_cpus
+	banned_cpus=$(sed -n 's/^IRQBALANCE_BANNED_CPUS=\"\?\([^\"]*\)\"\?/\1/p' "$IRQBALANCE_CONF")
+	if [ -z "$banned_cpus" ]; then
+		echo "error: unable to read 'IRQBALANCE_BANNED_CPUS' from ${IRQBALANCE_CONF}"
+		return 2
+	fi
+
+	[ "$expected_banned_cpus" == "$banned_cpus" ]
```

Comment:
if the restore file is disabled, it should not be created implicitely, right?
if so, let's add another test which checks this flow. Basically, let's split this test in two:
1. with restore-file disabled, if the file is missing, it is NOT consumed (obv.) and NOT created
2. with restore-file disabled, if the file is present, it is NOT consumed

Reply:
done